<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1247</width>
    <height>676</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout_2">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="tabPosition">
       <enum>QTabWidget::West</enum>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="mainTab">
       <attribute name="title">
        <string>Визуализатор</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="ALiveArenaWidget" name="liveArenaWidget" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="graphTab">
       <attribute name="title">
        <string>Графики</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_4">
        <item>
         <widget class="AChart" name="mainChart" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="statTab">
       <attribute name="title">
        <string>Статистика</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="ABestSurvivedWiget" name="bestSurvivedWidget" native="true"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="playButton">
        <property name="text">
         <string>Пропустить поколений</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="generationCountEdit">
        <property name="placeholderText">
         <string>Кол-во поколений</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="startAutoButton">
        <property name="text">
         <string>Запустить жизнь</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLineEdit" name="timeActionEdit">
        <property name="placeholderText">
         <string>Время действия мСек.</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="stopAutoButton">
        <property name="text">
         <string>Пристановить жизнь</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="numGenerationLabel">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::Expanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1247</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Файл</string>
    </property>
    <addaction name="saveLiveObjectsAction"/>
    <addaction name="loadLiveObjectsAction"/>
   </widget>
   <widget class="QMenu" name="menu_2">
    <property name="title">
     <string>Мир</string>
    </property>
    <addaction name="setBorderAction"/>
    <addaction name="unsetBorderAction"/>
   </widget>
   <widget class="QMenu" name="menu_3">
    <property name="title">
     <string>Настройки</string>
    </property>
    <addaction name="settingsChangeAction"/>
   </widget>
   <addaction name="menu"/>
   <addaction name="menu_2"/>
   <addaction name="menu_3"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="saveLiveObjectsAction">
   <property name="text">
    <string>Сохраниь лучших</string>
   </property>
  </action>
  <action name="loadLiveObjectsAction">
   <property name="text">
    <string>Загрузить лучших</string>
   </property>
  </action>
  <action name="setBorderAction">
   <property name="text">
    <string>Ограниченный</string>
   </property>
  </action>
  <action name="unsetBorderAction">
   <property name="text">
    <string>Свободный</string>
   </property>
  </action>
  <action name="settingsChangeAction">
   <property name="text">
    <string>Настройки</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ALiveArenaWidget</class>
   <extends>QWidget</extends>
   <header>alivearenawidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ABestSurvivedWiget</class>
   <extends>QWidget</extends>
   <header>abestsurvivedwiget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>AChart</class>
   <extends>QWidget</extends>
   <header>AChart/achart.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
